tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1137,6 +1137,9 @@ unsigned CGMSHLSLRuntime::ConstructStructAnnotation(DxilStructAnnotation *annota
         fieldAnnotation.SetPrecise();
     }
 
+    if (RD->isUnion())
+      break;
+
```

Comment:
If you put a union in a constant buffer, the size of the union comes back as 0, which isn't correct.

Reply:
Your comment above::
I don't think the struct annotations are correct, and one test that demonstrates it is adding a union to a ConstantBuffer.

Can you tell me which test case is this about?

Reply:
There isn’t a test case in the PR that covers this, but any shader using a `ConstantBuffer<T>`, where `T` is or contains a union type will generate incorrect cbuffer layout information.